### DISCUSSION TITLE
Federated mode and Local break-out ?

### DISCUSSION THREAD

**Discussion by ferrieux** [magma/magma#15318] [Q&A] [answered]
Hi,

We have configured Magma with inbound roaming to a PLMN, and would like to do local break-out, meaning:
 - user plane local to the AGW
 - control plane with remote HSS

However, we only have two options `local_subscriber` and `s8_subscriber` and none of them does the job:
 - `local_subscriber` does the local user plane but only queries the local subscriberdb for control
 - `s8_subscriber` does the remote HSS control, but also insists on having a remote S8 for user plane => no local break-out

Is there a way out ?

**Comment by panyogesh** [comment]
@ferrieux : This would need investigation. It seems it was [tried](https://magma.github.io/magma/docs/next/lte/dev_teid_allocation#local-subscriber-teid-allocation) before.  Can you please help us in identifying the priority. I can put a proposal to verify it in next release (post 1.9).

**Reply by ferrieux** [reply, marked as answer]
In the meantime, I found the solution, in the source: there is a third option named `SPGW_SUBSCRIBER` that does exactly that.

Note, this option  is **not** exposed in the configuration language, but can be nonetheless selected by  **omitting** the intended PLMN from the federated_mode_mapping->mapping list. For example, the config below sets PLMNs `00101`, `00102`, and `00103` respectively to LOCAL_SUBSCRIBER, S8_SUBSCRIBER, and SPGW_SUBSCRIBER: the latter by the mere **absence** of `00103` from the list.

```json
"federated_modes_mapping": {
    "enabled": true,
    "mapping": [
      {
         "mode": "local_subscriber",
         "plmn": "00101"
      },
      {
         "mode": "s8_subscriber",
         "plmn": "00102"
      }
    ]
}
```

You might decide to keep it that way, but It might be worth documenting this at least :)
A more user-friedly would be to define a new config value "spgw_subscriber". Your call.
